samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst

```diff
@@ -0,0 +1,297 @@
+This sample also demonstrates how to implement error tolerance in your cellular applications without relying on reboot loops.
+
+.. _nrf_cloud_mqtt_multi_services_sample_requirements:
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :sample-yaml-rows:
+
+.. _nrf_cloud_mqtt_multi_services_sample_features:
+
+Features
+********
+
+This sample implements or demonstrates the following features:
+
+* Error-tolerant use of the `nRF Cloud MQTT API`_ using the :ref:`nrf_modem <nrfxlib:nrf_modem>` and :ref:`nrf_cloud <lib_nrf_cloud>` libraries.
+* Support for `Firmware Over The Air (FOTA) <nRF Cloud Getting Started FOTA documentation_>`_ via the `nRF Cloud`_ portal.
+* Support for `modem AT Commands <AT Commands Reference Guide_>`_ over UART using the :ref:`AT Host library<lib_at_host>`.
+* Periodic cellular and GNSS location tracking using the :ref:`location library <lib_location>`.
+* Periodic temperature sensor sampling on your `Nordic Thingy:91`_, or fake temperature  measurements on your `Nordic nRF9160 DK`_.
+* Transmission of sensor and GNSS location samples to the `nRF Cloud`_ portal as `nRF Cloud Device Messages`_.
+* Construction of valid `nRF Cloud Device Messages_` using `cJSON`_.
+
+
+.. _nrf_cloud_mqtt_multi_services_sample_structure:
+.. _nrf_cloud_mqtt_multi_services_sample_structure_and_theory:
+.. _nrf_cloud_mqtt_multi_services_sample_structure_and_theory_of_operation:
+.. _nrf_cloud_mqtt_multi_services_sample_theory_of_operation:
+
+Structure and Theory of Operation
+*********************************
+
+This sample is separated into a number of smaller functional units.
+The top level functional unit and entry point for the sample is :file:`src/main.c`.
+This file starts three threads, each with a distinct function:
+
+* Connection Thread (``conn_thread``): Runs the `Connection Management Loop <nrf_cloud_mqtt_multi_services_sample_connection_management_loop_>`_, which maintains a connection to the `nRF Cloud MQTT API`_.
+* Application Thread (``app_thread``): Runs the `Main Application Loop <nrf_cloud_mqtt_multi_services_sample_main_application_loop_>`_, which may add `Device Messages <nRF Cloud Device Messages_>`_ to the `Device Message Queue <nrf_cloud_mqtt_multi_services_sample_device_message_queue_>`_.
+* Message Thread (``msg_thread``): Processes the `Device Message Queue <nrf_cloud_mqtt_multi_services_sample_device_message_queue_>`_ whenever there is an active connection.
+
+.. _nrf_cloud_mqtt_multi_services_sample_connection_management_loop:
+
+Connection Management Loop
+==========================
+
+The Connection Management Loop performs four primary tasks:
+
+1) Configure and activate the cellular modem.
+#) Start a connection to the LTE Network.
+#) Maintain a connection to the `nRF Cloud MQTT API`_.
```

Comment:
Hmm, yeah, you're not wrong -- I worry that being that specific may make the documentation harder to follow. The nice thing about mentioning the `nRF Cloud MQTT API` is that a reader can click on that reference. I don't think we have any specific docs for the MQTT broker in general.

I would make an analogy to the REST API -- If I claim to connect to a REST API, then technically that would be a connection to the HTTP server **_hosting_** the REST API, not to the API itself. It is generally understood that when I say "Connection to REST API", said connection will in actuality be to an HTTP server. But it is more communicative to mention the API, because the API is what will have Nordic-specific documentation, whereas the HTTP server architecture is usually not documented or even directly acknowledged in most cases

I think the same argument would apply here. What do you think?

Reply:
Maybe instead just say "maintain a connection to nRF Cloud"

Reply:
Sure, though in cases where I am talking about aspects very specific to the API itself, such as:

> If the connection to `nRF Cloud MQTT API`_ does not become ready within this timeout

I feel I should still mention the API, if that's alright with you